### DISCUSSION TITLE
Helm chart configuration to authenticate via Azure OAuth2

### DISCUSSION THREAD

**Discussion by ccontino84** [provectus/kafka-ui#3550] [Q&A] [answered]
Hi everybody,

I'm trying to configure kafka-ui to authenticate via OAuth2, but I can't find an example for Azure using the Helm chart as most of the documentation is about docker.
The kafka cluster is deployed using strimzi 0.34.0 and I'm able to connect to the cluster using the following command:
```bash
kafka-console-consumer --bootstrap-server kafka-bootstrap:9094 \
  --topic my_topic --group my_group \
  --consumer-property 'security.protocol=SASL_SSL' \
  --consumer-property 'sasl.mechanism=OAUTHBEARER' \
  --consumer-property 'sasl.jaas.config=org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required clientId="xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx" clientSecret="***" scope="https://management.core.windows.net/.default";' \
  --consumer-property 'sasl.login.callback.handler.class=org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerLoginCallbackHandler' \
  --consumer-property 'sasl.oauthbearer.token.endpoint.url=https://login.microsoftonline.com/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/oauth2/v2.0/token'
```
How can I  translate this in the YAML configuration?
Thanks

**Comment by Haarolean** [comment]
Hi, please take a look [here](https://docs.kafka-ui.provectus.io/configuration/authentication/oauth2).

**Reply by ccontino84** [reply]
Hi, thank you for the reply, but that doesn't help me a lot. There's no example using Azure in the documentation and I don't know exactly how to use the parameters that I set in the command line.

**Reply by Haarolean** [reply]
@ccontino84 well it seems like you need to authenticate into kafka itself, oauth is not really related here.

**Reply by ccontino84** [reply]
I know how to authenticate using the command line, what I don't know is how to use the Helm chart to do the same

**Comment by Haarolean** [comment]
Please take a look at this [example](https://github.com/provectus/kafka-ui/blob/master/documentation/compose/kafka-ui-sasl.yaml#L15), you'll get an idea how to format your properties.

**Reply by ccontino84** [reply]
I'd like to understand how to translate the docker configuration into the Helm chart, sorry I couldn't find enough documentation about that

**Reply by Haarolean** [reply, marked as answer]
@ccontino84 take a look [here](https://github.com/provectus/kafka-ui/tree/master/charts/kafka-ui). You can either use a yaml config map (I strongly recommend yaml over env properties) or env properties way.
Here's some example of a config map with [env vars](https://github.com/provectus/kafka-ui/blob/issue/doc-3142/configuration/kafka-ui-ssl.md).

**Reply by ccontino84** [reply]
Thank you, I'll try your suggestions and let you know if I still have issues :+1:
